catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/filesystem/FileSystemType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.catalog.lakehouse.paimon.filesystem;
+
+public enum FileSystemType {
+  LOCAL_FILE,
+  HDFS,
+  S3;
+
+  public static FileSystemType fromString(String type) {
+    for (FileSystemType fileSystemType : FileSystemType.values()) {
+      if (fileSystemType.name().equalsIgnoreCase(type)) {
+        return fileSystemType;
+      }
+    }
+
+    throw new IllegalArgumentException("Unsupported file system type: " + type);
+  }
+
+  public static FileSystemType fromStoragePath(String storagePath) {
+    if (storagePath.startsWith("s3://")) {
```

Comment:
shoud we consider upper cases like `S3://`?

Reply:
HDFS only support lower case 'hdfs', if we use `HDFS://xxx`

```text
Caused by: MetaException(message:Got exception: java.io.IOException No FileSystem for scheme: HDFS)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_database_result$create_database_resultStandardScheme.read(ThriftHiveMetastore.java:26660)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_database_result$create_database_resultStandardScheme.read(ThriftHiveMetastore.java:26628)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_database_result.read(ThriftHiveMetastore.java:26562)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:88)
```